### PR TITLE
Fixes #702

### LINKED ISSUE #702: Modeshape Backend for Web Editor

### PR DESCRIPTION
**Pull request by shiv12095**
Signed-off-by: shiv12095 <shiv12095@iiitd.ac.in>

### CODE REVIEW

File: server/devtool/projectrepository-api/src/main/java/org/eclipse/vorto/devtool/projectrepository/IProjectRepositoryService.java [hunk 1]
```diff
@@ -14,14 +14,16 @@
...
+
 import java.util.Map;
 
-import org.eclipse.vorto.devtool.projectrepository.model.FileResource;
-import org.eclipse.vorto.devtool.projectrepository.model.FolderResource;
-import org.eclipse.vorto.devtool.projectrepository.model.ProjectResource;
-import org.eclipse.vorto.devtool.projectrepository.model.Resource;
-import org.eclipse.vorto.devtool.projectrepository.model.ResourceContent;
-import org.eclipse.vorto.devtool.projectrepository.model.UploadHandle;
+import org.eclipse.vorto.devtool.projectrepository.exception.ResourceAlreadyExistsError;
```
Comment (aedelmann) [inline server/devtool/projectrepository-api/src/main/java/org/eclipse/vorto/devtool/projectrepository/IProjectRepositoryService.java]: package names should be singular not plural, e.g model vs models.

File: server/devtool/projectrepository-modeshape/src/main/java/org/eclipse/vorto/devtool/projectrepository/modeshape/utils/ProjectRepositoryModeshapeConstants.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.vorto.devtool.projectrepository.utils.ProjectRepositoryConstants;
+
+public class ProjectRepositoryModeshapeConstants {
+
+	public static final String VORTO = "vorto";
```
Comment (aedelmann) [inline server/devtool/projectrepository-modeshape/src/main/java/org/eclipse/vorto/devtool/projectrepository/modeshape/utils/ProjectRepositoryModeshapeConstants.java]: Constant classes are somewhat "ugly". Usually constants belong to a certain class. From what I can see, most of your defined constants are used by the Modeshape repository implementation. Why not defining them there ?

Comment (shiv12095) [reply server/devtool/projectrepository-modeshape/src/main/java/org/eclipse/vorto/devtool/projectrepository/modeshape/utils/ProjectRepositoryModeshapeConstants.java]: The constants are being used in multiple files, Modeshape repository implementation and Resource query modeshape implementation. 
That is the main reason I moved them to a separate constant file. 

As for the code being "ugly", I can look up some better coding practices to keep the constants and make the code more readable

File: server/devtool/webeditor/pom.xml [hunk 3]
```diff
@@ -183,7 +177,39 @@
...
+		<dependency>
+			<groupId>org.modeshape</groupId>
+			<artifactId>modeshape-jcr-api</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.modeshape</groupId>
+			<artifactId>modeshape-common</artifactId>
+		</dependency>
+        <dependency>
+            <groupId>org.modeshape</groupId>
```
Comment (aedelmann) [inline server/devtool/webeditor/pom.xml]: Is this dependency to webdav required ?

Comment (shiv12095) [reply server/devtool/webeditor/pom.xml]: No, that is not need
Added that during testing, forgot to remove that

**Pull request merged**